🔨 chore: add docker compose to setup local services for development

Asuka109 opened the pull request:
#### 💻 变更类型 | Change Type

<!-- For change type, change [ ] to [x]. -->

- [ ] ✨ feat
- [ ] 🐛 fix
- [ ] ♻️ refactor
- [ ] 💄 style
- [x] 👷 build
- [ ] ⚡️ perf
- [ ] 📝 docs
- [x] 🔨 chore

#### 🔀 变更说明 | Description of Change

This PR can helps to setup local development environment for the lobe server mode. With it you can do so by following steps:

```bash
$ cat .env.development >> .env
$ docker-compose up -d
$ pnpm db:migrate
```

And then just execute the `pnpm dev` command and it will load environement variables from the `.env` file. Which is a temporarily untracked file and only be used for local development.

Will update documentations later.

#### 📝 补充信息 | Additional Information

<!-- Add any other context about the Pull Request here. -->

## Summary by Sourcery

Add Docker Compose setup to orchestrate local development services

Build:
- Introduce root docker-compose.yml defining services for local development (network-service, postgresql, minio, casdoor, searxng), volumes, and network
- Add .env.development template for environment variables
- Include .gitignore entry for docker-compose local overrides

Asuka109 committed:
👷 build: add docker compose to setup local services for development

Asuka109 commented:
Will close https://github.com/lobehub/lobe-chat/pull/8578

arvinxx commented on docker-compose.yml:
prefer name as `docker-compose.development.yml`

Asuka109 replied:
sure

Asuka109 committed:
👷 build: setup dotenv-expand for all the npm scripts

Asuka109 committed:
Merge branch 'main' into feat/local-server-services

Asuka109 committed:
🐛 fix: remove useless comments

Asuka109 commented on .npmrc:
```diff
@@ -4,6 +4,9 @@ resolution-mode=highest
 enable-pre-post-scripts=true
 
+# Load dotenv files for all the npm scripts
+node-options="--require dotenv-expand/config"
```
Yes I believe that. It should works on any nodejs-compatible runtime with npm-compatible package manager. Including bun I think. But I didn't found the documentation to describe the behavior.

Asuka109 replied:
Btw I used dotenv expand because of the template string syntax. It can works with the new .env.development file.

arvinxx commented:
please reslove the conflict and the left is LGTM

Asuka109 committed:
Merge branch 'main' into feat/local-server-services

Asuka109 committed:
📚 docs: add server-side database setup guide with i18n support

- Add comprehensive server-side database setup documentation
- Include step-by-step Docker service configuration
- Add Chinese localization for better developer experience
- Cover environment setup, database migration, and verification steps

Asuka109 committed:
Merge branch 'main' into feat/local-server-services

arvinxx committed:
Merge branch 'main' into feat/local-server-services

arvinxx reviewed (approved):
lgtm now
